docs/superchain-paymaster/README.md

```diff
@@ -0,0 +1,158 @@
+# 💳 Superchain paymaster
+
+## Overview
+
+The Superchain paymaster is a ERC-4337 [verifying paymaster](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/samples/VerifyingPaymaster.sol) that sponsors transactions for smart accounts on the Superchain.
+
+## Endpoints
+
+| Network      | Chain ID  | Endpoint                                       |
+| ------------ | --------- | ---------------------------------------------- |
+| Sepolia      | 11155111  | https://paymaster.optimism.io/v1/11155111/rpc  |
+| OP Sepolia   | 11155420  | https://paymaster.optimism.io/v1/11155420/rpc  |
+| Base Sepolia | 84532     | https://paymaster.optimism.io/v1/84532/rpc     |
+| Zora Sepolia | 999999999 | https://paymaster.optimism.io/v1/999999999/rpc |
+| Mode Sepolia | 919       | coming soon                                    |
+
+Mainnet endpoints coming soon!
+
+## Supported methods
+
+### pm_sponsorUserOperation
+
+This endpoint returns gas estimation, fee estimation, and paymaster signature for a [verifying paymaster](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/samples/VerifyingPaymaster.sol). Gas estimation and fee estimation params are included to reduce number of RPC calls necessary when sending a user operation.
+
+**Request**
+
+```typescript
+{
+	jsonrpc: "2.0",
+	id: 1,
+	method: "pm_sponsorUserOperation",
+	params: [
+		{
+			sender, // (required) address in hex string
+			nonce, // (required) hex string
+			initCode, // (required) hex string, if account is already deployed, pass in "0x"
+			callData, // (required) hex string
+			paymasterAndData, // (optional) hex string, will be ignored
+			signature, // (required) hex string, can be a dummy signature that doesn't revert
+			maxFeePerGas, // (optional), will override fee estimations done by the paymaster
+			maxPriorityFeePerGas, // (optional) will override fee estimations done by the paymaster
+			callGasLimit,  // (optional) hex string, will be ignored
+			verificationGasLimit, // (optional) hex string, will be ignored
+			preVerificationGas, // (optional) hex string, will be ignored
+		},
+		"0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789", // entrypoint address (required)
+	],
+
+}
+
+```
+
+**Response**
+
+```typescript
+{
+	jsonrpc: "2.0",
+	id: 1,
+	result: {
+		paymasterAndData: "0x...", // paymaster signature
+		callGasLimit: "0x...",  // gas estimation
+		verificationGasLimit: "0x...", // gas estimation
+		preVerificationGas: "0x...", // gas estimation
+		maxFeePerGas: "0x...", // fee estimation
+		maxPriorityFeePerGas: "0x...", // fee estimation
```

Comment:
are there docs we can point to that explain what each of these fields should be used for?

Reply:
good point. will add a line about adding this to ur user operation